### PR TITLE
docs: add process doc for mockoon usage

### PR DESCRIPTION
**Pull request by LakshayaT**
**Description:**
This PR adds process documentation for Mockoon with detailed usage instructions and screenshots. This documentation will serve as a guide for current and future developers to understand and leverage Mockoon for their projects. 

**Related changes:**
- Create a mockoon-usage.md doc in new process folder
- Create an assets folder for adding an images that would be used
- Add an introduction section discussing the problem, current process pipeline, and the solution
- Add brief section about why use mockoon, with steps and mockoon pipeline
- Add a small setup section to link to the download page for mockoon
- Create a detailed usage section with sub-sections like understanding ui, environments creation and settings, adding and configuring routes, copying route and environment configurations etc
- Create another section for advanced usage topics such as templating, data buckets, crud routes, proxy mode, logs, callbacks etc
- Add a reference to the official mockoon documentation
- Add screenshots for better explaining steps to be followed for certain tasks as required

## Summary by CodeRabbit

- **Documentation**
	- Updated the `process/mockoon-usage.md` file with a detailed guide on utilizing Mockoon for running mock APIs locally, covering setup, configuration, advanced features like templating, CRUD routes, proxy mode, and requests logging.

### CODE REVIEW

File: process/mockoon-usage.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+Alternatively, you can directly copy the configuration by right clicking the environment and selecting **Copy configuration to clipboard (JSON)**.
+
+This content can then be pasted in a new file if needed and shared with the frontend developer(s).
+
+### Opening Environments
+
+You can open environments from JSON file on your system as follows:
+
+1. Select **File** in the top menu bar
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md:131]: "File" is a button in the UI, and is capitalised as such. We are referring to that button in this document, so need to use "File" and no "file".

File: process/mockoon-usage.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+There are two possible ways to avoid delays due to the above pipeline:
+
+- Provide API details to the frontend before development. We can try to use AI tools to do this, but it was found that these are not reliable. This will also require human intervention for testing the APIs, for example, using different responses.
+- Use a mock API server. This will mock the API responses using static or random responses as configured.
+
+[Back to top](#mockoon-usage)
+
+## 2. Why Mockoon?
+
+[Mockoon](https://mockoon.com) is a free and open source application that is cross platform and provides an easy and quick way to run mock APIs locally. There is no mandatory login or account requirement for using this.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+![Current Pipeline](assets/mockoon-without-pipeline.png)
+
+This pipeline requires the backend developer to always be one step ahead of the frontend developer in development of a particular module.
+
+### The Solution
+
+There are two possible ways to avoid delays due to the above pipeline:
+
+- Provide API details to the frontend before development. We can try to use AI tools to do this, but it was found that these are not reliable. This will also require human intervention for testing the APIs, for example, using different responses.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+### Creating New Environments
+
+Mockoon allows you to create isolated environments for your mock APIs. Each environment can have its own routes, settings, and responses.
+
+1. Launch the Mockoon application.
+2. Create a new environment:
+
+   1. Click on the **New environment** button. This can be found near the top of the Environments list.
+   2. Provide a descriptive name for your environment file and create it.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+
+- **API URL:** Base URL, port, prefix
+- **TLS:** Enabling and setting up TLS for HTTPS support
+- **Global Latency:** Delay (latency) for all responses in the environment
+
+### Viewing and Copying Environment Configuration
+
+You can view and copy environment configuration as follows:
+
+1. Right click on the target environment
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+2. Click on **Show data file in explorer/finder**
+3. At this point, you can copy the JSON file or open it and copy the contents.
+
+![Mockoon view/copy environment config](assets/mockoon-viewcopy-environment-config.png)
+
+Alternatively, you can directly copy the configuration by right clicking the environment and selecting **Copy configuration to clipboard (JSON)**.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+Add new HTTP routes using the following steps:
+
+1. Select the environment in which you want to add the route
+2. In the **Routes** tab, click on the **+** icon and select **HTTP route**
+
+![Mockoon new HTTP route](assets/mockoon-new-http-route.png)
+
+Configure the added route with various parameters such as:
+
+- **Method:** Select corresponding method from the dropdown such as GET, POST, etc
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+### Copying and Adding Route Configuration
+
+You can copy a route configuration following these steps:
+
+1. Select the target route, and right-click on it
+2. Click on **Copy configuration to clipboard (JSON)**
+
+![Mockoon copy route configuration](assets/mockoon-copy-route-configuration.png)
+
+You can then paste this content in a file for saving it.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+
+1. Copy the route configuration via the above steps or from a file
+2. Click on **Routes** in the top menubar and select **Add route from clipboard**
+
+![Mockoon add route from clipboard](assets/mockoon-add-route-from-clipboard.png)
+
+### Toggling and Deleting Routes
+
+- In case specific routes are temporarily required to be disabled, it can be done so by right-clicking on the route and selecting **Toggle**
+- In order to delete a route, right-click on the route and click on **Delete** and then **Confirm deletion**
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+- **Path:** Define the CRUD route path
+- **Data:** Select the data bucket to be used. You can also introduce response latency for the selected response
+- **Headers:** Add custom headers as and if needed, defining the name and value
+- **Modes (Random, Sequential, Disabled Rules, Fallback):** Choose how responses are selected by selecting the mode beside the responses dropdown
+
+Refer to the official Mockoon documentation to learn more about the operations allowed in CRUD routes.
+
+### Using Proxy Mode
+
+Mockoon supports partial mocking of an API endpoints by forwarding the requests that does not match a declared route to the URL of your choice. This can, for example, be used for using already developed and deployed APIs.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/b39269f804c7670594d17b37165712f4d9647198

File: process/mockoon-usage.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: If I remove one space before the indented list items, then those items become part of the parent list. I want to keep them nested under the main list's item only.

File: process/mockoon-usage.md [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+
+Refer to the [GUI Cheat Sheet](https://mockoon.com/docs/latest/gui-cheat-sheet/) for reviewing and understanding the different user interface elements in the application.
+
+### Creating New Environments
+
+Mockoon allows you to create isolated environments for your mock APIs. Each environment can have its own routes, settings, and responses.
+
+1. Launch the Mockoon application.
+2. Create a new environment:
+   1. Click on the **New environment** button. This can be found near the top of the Environments list.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md:95]: As mentioned in https://github.com/OsmosysSoftware/dev-standards/pull/49#discussion_r1514517333, doing so will break the list hierarchy that I am trying to maintain. Rendering with 2 spaces in Github does not indent the list item, thus the usage of 3 spaces.

File: process/mockoon-usage.md [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+- In case specific routes are temporarily required to be disabled, it can be done so by right-clicking on the route and selecting **Toggle**
+- To delete a route, right-click on the route and click on **Delete** and then **Confirm deletion**
+
+![Mockoon toggle/delete routes](assets/mockoon-toggledelete-routes.png)
+
+[Back to top](#mockoon-usage)
+
+## 5. Advanced Usage
+
+This section outlines some of the advanced features that Mockoon offers which might not be used regularly, but can come in handy from time-to-time.
```
[the remark this answers is not in the document]
Comment (LakshayaT) [reply process/mockoon-usage.md]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/2746f40371a1a1d7dc5d250bfdaba3911b162f07

File: process/mockoon-usage.md [hunk 13]
```diff
@@ -0,0 +1,317 @@
+# Mockoon Usage
```
Comment (xixas) [inline process/mockoon-usage.md:1]: @LakshayaT  Use this link to generate better quality pipeline images. 
https://lucid.app/lucidchart/a653ad15-e662-4102-8739-036b6d77bd6a/edit?viewport_loc=-1101%2C16%2C1930%2C1032%2C0_0&invitationId=inv_1ede69a0-72b4-474a-b991-eaf9c1c60744

File: process/mockoon-usage.md [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+
+## 3. Setup
+
+Go to the [Mockoon website](https://mockoon.com) and head to the [download page](https://mockoon.com/download/#download-section). Download and install the required package as per your OS.
+
+Take the tour to gain an understanding of the various UI elements and features.
+
+[Back to top](#mockoon-usage)
+
+## 4. Usage
```
Comment (xixas) [inline process/mockoon-usage.md:81]: In the usage.
- You need to mention what are expected steps from Backend Guys.
-  And what are the expected actions and features to be used by frontend guys(like playing with latency, responses Sequential/Random Response modes) 

Specify importance of  topics like adding different responses, latency(global and local), Sequential/Random Response Modes and Proxy Server. And next would be templating.

Keep others topics separate so Importance of usage will be clearer. Also specify about certain features that they wont be used(Example CRUD feature is useless for our process, so we wont want people spending time on it, but just know of the feature).

**Review by xixas — CHANGES_REQUESTED** [review]

File: process/mockoon-usage.md (hunk 13 above)
Comment (LakshayaT) [reply process/mockoon-usage.md:1]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/2beeb35f34d00bdc66e5a43ec2d625671ae8227a

File: process/mockoon-usage.md (hunk 14 above)
Comment (LakshayaT) [reply process/mockoon-usage.md:84]: Addressed in https://github.com/OsmosysSoftware/dev-standards/pull/49/commits/78438cdadc6d635ecc757d32148cb97f91835fd7

**Review by xixas — APPROVED** [review]

**Pull request merged**